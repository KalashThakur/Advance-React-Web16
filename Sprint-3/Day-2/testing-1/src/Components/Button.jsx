import React from 'react';
import "./Button.css";

export const Button = ({children, variant, colorScheme, onClick}) => {
  return (
    <button data-testid="cButton" className={`btn ${colorScheme || "red"} ${variant || "ghost"}`} onClick={onClick}>
        {children}
    </button>
  )
}
