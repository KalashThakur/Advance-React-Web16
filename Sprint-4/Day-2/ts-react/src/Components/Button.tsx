import React from 'react';

interface ButtonProps {
    label: string;
    handleClick: () => void
}

// export const Button = (props: ButtonProps) => {
//     const {label, handleClick} = props;
export const Button = ( {label, handleClick}: ButtonProps) => {

  return (
    <div>
        <button onClick={handleClick}> {label} </button>
    </div>
  )
}
