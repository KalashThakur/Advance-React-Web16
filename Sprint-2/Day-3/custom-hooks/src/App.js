import './App.css';
import { Counter } from './Components/Counter';
import { Github } from './Components/Github';
import { Timer } from './Components/Timer';

function App() {
  return (
    <div className="App">
      {/* <Timer />
      <Github /> */}
      <Counter />
    </div>
  );
}

export default App;
