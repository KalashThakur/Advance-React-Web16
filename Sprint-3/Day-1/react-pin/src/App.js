import logo from "./logo.svg";
import "./App.css";
import { Pin } from "./Compo/Pin";
import { useState } from "react";

function App() {
  const [otp, setOtp] = useState("");
  return (
    <div className="App">
      <Pin
        length={4}
        onChange={(value) => {
          setOtp(value);
        }}
      />
      <h2>The value of OTP is {otp}</h2>
    </div>
  );
}

export default App;
