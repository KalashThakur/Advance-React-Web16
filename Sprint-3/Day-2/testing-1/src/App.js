import './App.css';
import { Button } from './Components/Button';
import { useState } from 'react';

function App() {
  const [theme, setTheme] = useState("light");

  return (
    <div className="App">
      <h3>Current Theme is {theme}</h3>
      <Button onClick={() => {setTheme(theme === "light" ? "dark" : "light")}}> Change Theme </Button>
    </div>
  );
}

export default App;
