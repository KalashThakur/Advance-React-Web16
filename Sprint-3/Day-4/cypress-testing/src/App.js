import useState from "react";
import './App.css';

function App() {

  const [count, setCount] = useState(0)
  return (
    <div className="App">
      <h2>Count is {count}</h2>
      <button onClick={(prev) => prev-1}>Dec</button>
      <button onClick={(prev) => prev+1}>Inc</button>
    </div>
  );
}

export default App;
