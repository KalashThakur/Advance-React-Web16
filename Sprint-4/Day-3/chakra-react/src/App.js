import logo from './logo.svg';
import './App.css';
import {Box, Container, extendTheme} from "@chakra-ui/react";
import { Form } from './Components/Form';
import { Layouts } from './Components/Layouts';





function App() {
  return (
    <Container border="1px solid red" maxWidth="2xl" >
      {/* <Form/> */}
      <Layouts />
    </Container>
  );
}

export default App;
