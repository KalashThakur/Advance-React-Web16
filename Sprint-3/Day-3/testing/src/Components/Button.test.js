import {fireEvent, render, screen} from "@testing-library/react";
import App from "../App";
import Button from "./Button";
import userEvent from "@testing-library/user-event";
import { createRenderer } from "react-dom/test-utils";
import renderer from "react-test-renderer";
//render test our component virtually, it will not really create the component in browser
//render in index.js create the dom structure in browser

describe("Button tests", () => {

    it("should have button in dom", () => {

        render(<Button>Click Me</Button>) //virtually
        // screen.debug(); //will show dom virtually in terminal
        let button = screen.getByText("Click Me") //screen is similar to document
        expect(button).toBeInTheDocument();
    });

    it("should have button with text Click Me in App.js", () => {
        render(<App></App>)
        let button = screen.getByTestId("cButton")
        expect(button).toBeInTheDocument();
    });

    it("should render empty button", () => {
        render(<Button></Button>);
        let button = screen.getByTestId("cButton");
        expect(button).toBeEmptyDOMElement();
    });

    it("should change theme", () => {
        render(<App></App>);
        let h3 = screen.getByText("Current Theme is light");
        expect(h3).toHaveTextContent("light");

        //after click on button, we again check h3
        let themeButton = screen.getByText("Change Theme");
        // userEvent.click(themeButton);
              //OR
        fireEvent.click(themeButton);
        expect(h3).toHaveTextContent("dark");
    });

    it("should call the given function", () => {
        const mockfn = jest.fn(); //we don't care how does this function work, we just want to check if our function onClick is triggered or not
        render(<Button onClick={mockfn}>Click Me</Button>);
        const btn = screen.getByTestId('cButton');
        fireEvent.click(btn);

        expect(mockfn).toBeCalled()
    });

    // Snapshot -> takes a screenshot of our code, represented in a particular format
    // it.only -> test only that test case
    it("should match snapshot", () => {
        const tree = renderer.create(<Button colorScheme={'green'} variant={'ghost'}>Add Counter</Button>).toJSON()
        expect(tree).toMatchSnapshot();
    });

    //basic utilities, hooks (not react hooks)

    // 1. beforeAll();
    // //before running test cases, this func will run first
    // //make a db call => output

    //2.  beforeEach();
    // //this func will run before every test case
    // //get the deployed link

    //3.  afterAll();

    //4.  afterEach();

})