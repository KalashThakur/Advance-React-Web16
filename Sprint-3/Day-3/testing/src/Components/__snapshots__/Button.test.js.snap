// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should match snapshot 1`] = `
<button
  className="btn green ghost "
  data-testid="cButton"
>
  Add Counter
</button>
`;
